Scripts/RemediationScripts/Remediate-EnableAADForKubernetesService.ps1

```diff
@@ -0,0 +1,564 @@
+                        @{Expression={$_.Location};Label="Location";Width=20;Alignment="left"}
+
+        $kubernetesServicesSkipped | Format-Table -Property $colsProperty -Wrap
+    }
+
+    if ($totalKubernetesServicesWithoutAADEnabled -eq 0)
+    {
+        Write-Host "No Kubernetes Service found with AAD disabled. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        break
+    }
+
+    Write-Host "Found $($totalKubernetesServicesWithoutAADEnabled) out of $($totalKubernetesServices) Kubernetes Service(s) with AAD disabled." -ForegroundColor $([Constants]::MessageType.Update)
+
+    # Back up snapshots to `%LocalApplicationData%'.
+    $backupFolderPath = "$([Environment]::GetFolderPath('LocalApplicationData'))\AzTS\Remediation\Subscriptions\$($context.Subscription.SubscriptionId.replace('-','_'))\$($(Get-Date).ToString('yyyyMMddhhmm'))\EnableAADForKubernetesServices"
+
+    if (-not (Test-Path -Path $backupFolderPath))
+    {
+        New-Item -ItemType Directory -Path $backupFolderPath | Out-Null
+    }
+
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "`n[Step 4 of 5] Backing up Kubernetes Service details..."
+
+    # Backing up Kubernetes Service details.
+    $backupFile = "$($backupFolderPath)\KubernetesClusterWithAADDisabled.csv"
+    $kubernetesServicesWithoutAADEnabled | Export-CSV -Path $backupFile -NoTypeInformation
+
+    Write-Host "Successfully backed up Kubernetes Service details to $($backupFolderPath)" -ForegroundColor $([Constants]::MessageType.Update)
+
+    if (-not $DryRun)
+    {
+        $colsProperty = @{Expression={$_.ResourceId};Label="Resource ID";Width=40;Alignment="left"},
+                        @{Expression={$_.ResourceGroupName};Label="Resource Group Name";Width=20;Alignment="left"},
+                        @{Expression={$_.ResourceName};Label="Resource Name";Width=20;Alignment="left"},
+                        @{Expression={$_.Location};Label="Location";Width=20;Alignment="left"},
+                        @{Expression={$_.isRBACEnabled};Label="Is RBAC enabled on Kubernetes Cluster?";Width=20;Alignment="left"},
+                        @{Expression={$_.isAADEnabled};Label="Is AAD enabled on Kubernetes Cluster?";Width=20;Alignment="left"}
+
+        Write-Host "`nFollowing Kubernetes cluster(s) are having AAD disabled:" -ForegroundColor $([Constants]::MessageType.Info)
+        $kubernetesServicesWithoutAADEnabled | Format-Table -Property $colsProperty -Wrap
+  
+        Write-Host $([Constants]::SingleDashLine)
+        Write-Host "AAD will be enabled for all Kubernetes Services."
+        Write-Host "Do you want to add Azure AD groups as administrators on each cluster?" -ForegroundColor $([Constants]::MessageType.Warning) -NoNewline
+        
+        $addAADGroup = $false
+        $input = Read-Host -Prompt "(Y|N)"
+
+        if($input -eq "Y")
+        {
+            $addAADGroup = $true
+        }
+        
+        Write-Host "RBAC must be enabled on Kubernetes cluster."
```

Comment:
Lets move this check above the AAD group check, as if the user don't want to enable then asking for AAD group choice does not make sense.

Reply:
Done.